### PR TITLE
feature: Record viewer

### PR DESCRIPTION
**Pull request by walnutdust**
Add a record viewer component to visualize the calls in a timeline

https://github.com/truera/trulens/assets/36098015/d982e73a-1ae6-4456-99e5-04f3cde78a88

Also part of this PR:
- Added component view types for trulens components like Feedbacks and utility classes like WithFeedbackFilterDocuments so they can better show up in the new timeline views.
- Wrapper for ThreadPoolExecutor in util.py that does our stack preservation across threads. Also changed WithFeedbackFilterDocuments to use this executor as the prior approach was not doing work in parallel like expected.

### CODE REVIEW

**Review by shayaks — APPROVED** [review]

File: trulens_eval/trulens_eval/app.py [hunk 1]
```diff
@@ -105,9 +107,9 @@ def class_is(cls: Class) -> bool:
         if cls.module.module_name.startswith("langchain."):
             return True
 
-        if any(base.module.module_name.startswith("langchain.")
-               for base in cls.bases):
-            return True
+        #if any(base.module.module_name.startswith("langchain.")
+        #       for base in cls.bases):
+        #    return True
```
Comment (rshih32) [inline trulens_eval/trulens_eval/app.py:112]: why is this removed?

File: trulens_eval/trulens_eval/app.py [hunk 2]
```diff
@@ -124,9 +126,9 @@ def class_is(cls: Class) -> bool:
         if cls.module.module_name.startswith("llama_index."):
             return True
 
-        if any(base.module.module_name.startswith("llama_index.")
-               for base in cls.bases):
-            return True
+        #if any(base.module.module_name.startswith("llama_index.")
+        #       for base in cls.bases):
+        #    return True
```
Comment (rshih32) [inline trulens_eval/trulens_eval/app.py:131]: why is this removed?

Comment (walnutdust) [reply trulens_eval/trulens_eval/app.py:131]: cc @piotrm0

**Review by rshih32 — APPROVED** [review]
